source/val/validate_image.cpp

```diff
@@ -750,12 +750,14 @@ spv_result_t ValidateTypeImage(ValidationState_t& _, const Instruction* inst) {
   if (spvIsVulkanEnv(target_env)) {
     if ((!_.IsFloatScalarType(info.sampled_type) &&
          !_.IsIntScalarType(info.sampled_type)) ||
-        (32 != _.GetBitWidth(info.sampled_type) &&
-         (64 != _.GetBitWidth(info.sampled_type) ||
-          !_.HasCapability(SpvCapabilityInt64ImageEXT)))) {
-      return _.diag(SPV_ERROR_INVALID_DATA, inst)
-             << "Expected Sampled Type to be a 32-bit int or float "
-                "scalar type for Vulkan environment";
+        ((32 != _.GetBitWidth(info.sampled_type)) &&
```

Comment:
Why did you change the logic? It looked correct previously, just the error message seemed out of date.

Reply:
hmm... I "ment" to change the logic, I realize what I am missing is a test with both a `Float64` and `Int64` as it would have a `Int64ImageEXT` capability and make it through

Reply:
should be fixed in updated commit now, move capability check out to Universal scope